### ISSUE TITLE
Option to disable integration managers server-wide (regardless of client used to connect) … [GDPR]

### ISSUE THREAD

**Issue by fooness** [matrix-org/synapse#8109]
We would like (and need) to disable integration managers on our server.

That’s no problem in the Element-Web we serve, but users could always connect with other Matrix clients we don’t have any control over. With those clients, users could add widgets from element web services, which would raise questions in regards to GDPR, privacy, whatever … in our institution, we’re basically not allowed to use external services.

In #matrix:matrix.org, @tulir pointed out that maybe blocking requests to the openid endpoint `` would solve this issue, but as we’re using a federation whitelist, we would need to block openid without disabling federation.

Also, maybe there’s a reason we might need openid for something else? (I don’t know.) I just want to say, maybe there can “simply” be a proper option/setting for synapse which would allow us to disabled the use of integration managers on our server (or, in other words, in rooms created from/on our server).

Hopefully I was able to phrase everything correctly and people can understand what we would need and why.

As always, thank you.

**Comment by clokep** [comment]
> maybe blocking requests to the openid endpoint `/_matrix/federation/v1/openid/` would solve this issue

I believe that this would only help with disabling integrations that require user information, some integrations are unauthenticated.

I'm struggling a bit to see how the homeserver is involved here. It seems like this might be:
* A spec issue (https://github.com/matrix-org/matrix-doc/) for room state having an option to disable integrations / widgets. (Or only allow them from certain integration managers or something.) This may or may not be possible with [MSC1957](matrix-org/matrix-doc#1957)'s `.well-known` publishing.
* A client issue to avoid having integrations / widgets.
* Solved at a networking level to block external services. (If you really can't use external services, you already must be doing this, right?)

I'm also a bit confused at how you're federating but not able to use external services. I'm guessing you're using a closed federation with only certain other servers, but it isn't really clear.

**Comment by tulir** [comment]
> I believe that this would only help with disabling integrations that require user information, some integrations are unauthenticated.

The integration manager is authed, so disabling OpenID should prevent users from using the default integration manager, which is what I think @fooness wants to do.

**Comment by fooness** [comment]
> The integration manager is authed, so disabling OpenID should prevent users from using the default integration manager, which is what I think @fooness wants to do.

That’s right. We’re basically looking for a way to prevent users on our server, independet of client, to use/access the integration manager.
